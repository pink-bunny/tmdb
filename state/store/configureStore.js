import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';

import reducer from './reducer';

const loggerMiddleware = (store) => (next) => (action) => {
  /* eslint-disable no-console */
  console.group(action.type);
  console.info('dispatching', action);
  const result = next(action);
  console.log('next state', store.getState());
  console.groupEnd();
  /* eslint-enable */

  return result;
};

export default function configureStore() {
  const middlewares = [loggerMiddleware, thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(reducer, undefined, composeWithDevTools(middlewareEnhancer));

  return store;
}
