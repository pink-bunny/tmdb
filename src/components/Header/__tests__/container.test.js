import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import ConnectedHeader from '../container';

describe('Header container', () => {
  const initialState = {
    session: {
      sessionId: '123456',
      username: 'John Doe'
    }
  };
  const store = configureStore()(initialState);
  const wrapper = shallow(<ConnectedHeader store={store} />).dive();

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
