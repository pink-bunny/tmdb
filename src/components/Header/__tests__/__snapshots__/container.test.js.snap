// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header container matches snapshot 1`] = `
<Header
  completeSession={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  userLoggedIn={true}
  username="John Doe"
/>
`;
